### PR TITLE
[Keyboard] add rartlice

### PR DESCRIPTION
**Pull request by alabahuy**
## Description

add new keyboard rartlice

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/rart/rartlice/config.h [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+#pragma once
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0x414C
+#define PRODUCT_ID      0x0065
+#define DEVICE_VER      0x0001
+#define MANUFACTURER    Alabahuy
+#define PRODUCT         Rartlice
+#define DESCRIPTION     65% Mechanical keyboard with Ergo layout
```
Comment (fauxpark) [inline keyboards/rart/rartlice/config.h:26]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/rart/rartlice/config.h (hunk 1 above)
Comment (alabahuy) [reply keyboards/rart/rartlice/config.h]: Done for update

**Comment by alabahuy** [comment]
@zvecr @fauxpark any update for this PR?

File: keyboards/rart/rartlice/rules.mk [hunk 2]
```diff
@@ -0,0 +1,25 @@
+# MCU name
+MCU = STM32F103
+
+# GENERIC STM32F103C8T6 board - stm32duino bootloader
+OPT_DEFS     = -DCORTEX_VTOR_INIT=0x2000
```
Comment (fauxpark) [inline keyboards/rart/rartlice/rules.mk:5]:
```suggestion
```

File: keyboards/rart/rartlice/rules.mk [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+MCU_LDSCRIPT = STM32F103x8_stm32duino_bootloader
+BOARD        = STM32_F103_STM32DUINO
+
+DFU_ARGS        = -d 1eaf:0003 -a2 -R
+DFU_SUFFIX_ARGS = -v 1eaf -p 0003
```
Comment (fauxpark) [inline keyboards/rart/rartlice/rules.mk:10]: #10558

(just a note to self if your PR gets merged first)

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/rart/rartlice/rules.mk (hunk 3 above)
Comment (alabahuy) [reply keyboards/rart/rartlice/rules.mk:10]: yes i will remove this and change to stm32Duino, waiting your PR merged

File: keyboards/rart/rartlice/rules.mk (hunk 2 above)
Comment (alabahuy) [reply keyboards/rart/rartlice/rules.mk]: Done

**Comment by alabahuy** [comment]
update from #10558 done @fauxpark

**Comment by alabahuy** [comment]
any update @fauxpark @zvecr @tzarc ?

File: keyboards/rart/rartlice/rules.mk [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+MOUSEKEY_ENABLE = yes   # Mouse keys
+EXTRAKEY_ENABLE = yes   # Audio control and System control
+CONSOLE_ENABLE = yes    # Console for debug
+COMMAND_ENABLE = yes    # Commands for debug and configuration
+SLEEP_LED_ENABLE = yes  # Breathing sleep LED during USB suspend
+NKRO_ENABLE = yes	    # USB Nkey Rollover
+BACKLIGHT_ENABLE = no
+RGBLIGHT_ENABLE = yes
+WS2812_DRIVER = spi
+OLED_DRIVER_ENABLE = yes
```
Comment (fauxpark) [inline keyboards/rart/rartlice/rules.mk:17]:
```suggestion
# Build Options
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = no       # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = yes        # Console for debug
COMMAND_ENABLE = yes        # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = yes      # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = yes           # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
WS2812_DRIVER = spi
OLED_DRIVER_ENABLE = yes
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/rart/rartlice/rules.mk (hunk 4 above)
Comment (alabahuy) [reply keyboards/rart/rartlice/rules.mk]: done

**Review by fauxpark — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Pull request merged**